File: frontend/test_simple_frontend.dart
/**
 * En basit frontend testi - Dart ile
 * Flutter yüklü olmadan da çalışır
 */

// Basit test framework'ü
class TestFramework {
  static int _passedTests = 0;
  static int _failedTests = 0;
  
  static void expect(dynamic actual, dynamic expected, String testName) {
    if (actual == expected) {
      _passedTests++;
      print('✅ $testName: PASSED');
    } else {
      _failedTests++;
      print('❌ $testName: FAILED');
      print('   Expected: $expected');
      print('   Actual: $actual');
    }
  }
  
  static void expectTrue(bool condition, String testName) {
    if (condition) {
      _passedTests++;
      print('✅ $testName: PASSED');
    } else {
      _failedTests++;
      print('❌ $testName: FAILED');
    }
  }
  
  static void expectFalse(bool condition, String testName) {
    if (!condition) {
      _passedTests++;
      print('✅ $testName: PASSED');
    } else {
      _failedTests++;
      print('❌ $testName: FAILED');
    }
  }
  
  static void printResults() {
    print('\n' + '=' * 50);
    print('📊 TEST RESULTS:');
    print('✅ Passed: $_passedTests');
    print('❌ Failed: $_failedTests');
    print('📈 Total: ${_passedTests + _failedTests}');
    print('=' * 50);
    
    if (_failedTests == 0) {
      print('🎉 All tests passed!');
    } else {
      print('⚠️ Some tests failed!');
    }
  }
}

// TerraMind Frontend Test Sınıfları
class Product {
  final String id;
  final String name;
  final String category;
  final String description;
  
  Product({
    required this.id,
    required this.name,
    required this.category,
    required this.description,
  });
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'description': description,
    };
  }
  
  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      description: json['description'],
    );
  }
}

class ProductService {
  static final ProductService _instance = ProductService._internal();
  factory ProductService() => _instance;
  ProductService._internal();
  
  final List<Product> _products = [
    Product(
      id: '1',
      name: 'wheat',
      category: 'Tahıl',
      description: 'Temel besin maddesi olan tahıl ürünü',
    ),
    Product(
      id: '2',
      name: 'corn',
      category: 'Tahıl',
      description: 'Yüksek verimli tahıl ürünü',
    ),
    Product(
      id: '3',
      name: 'cotton',
      category: 'Endüstriyel',
      description: 'Tekstil endüstrisi için lif ürünü',
    ),
  ];
  
  List<Product> getAllProducts() {
    return List.from(_products);
  }
  
  Product? getProductById(String id) {
    try {
      return _products.firstWhere((product) => product.id == id);
    } catch (e) {
      return null;
    }
  }
  
  List<Product> searchProducts(String query) {
    return _products.where((product) => 
      product.name.toLowerCase().contains(query.toLowerCase()) ||
      product.category.toLowerCase().contains(query.toLowerCase())
    ).toList();
  }
  
  List<Product> getProductsByCategory(String category) {
    return _products.where((product) => 
      product.category == category
    ).toList();
  }
}

class UserService {
  static final UserService _instance = UserService._internal();
  factory UserService() => _instance;
  UserService._internal();
  
  Map<String, String> _users = {
    'test@example.com': 'password123',
    'admin@example.com': 'admin123',
  };
  
  bool login(String email, String password) {
    if (_users.containsKey(email) && _users[email] == password) {
      return true;
    }
    return false;
  }
  
  bool register(String email, String password) {
    if (_users.containsKey(email)) {
      return false; // User already exists
    }
    _users[email] = password;
    return true;
  }
}

// Test Fonksiyonları
void testProductService() {
  print('\n🧪 Testing ProductService...');
  
  final productService = ProductService();
  
  // Test 1: GetAllProducts
  final allProducts = productService.getAllProducts();
  TestFramework.expect(allProducts.length, 3, 'GetAllProducts should return 3 products');
  
  // Test 2: GetProductById
  final product = productService.getProductById('1');
  TestFramework.expectTrue(product != null, 'GetProductById should return a product');
  TestFramework.expect(product?.name, 'wheat', 'Product name should be wheat');
  
  // Test 3: SearchProducts
  final searchResults = productService.searchProducts('wheat');
  TestFramework.expect(searchResults.length, 1, 'SearchProducts should return 1 result for wheat');
  
  // Test 4: GetProductsByCategory
  final grainProducts = productService.getProductsByCategory('Tahıl');
  TestFramework.expect(grainProducts.length, 2, 'GetProductsByCategory should return 2 grain products');
}

void testUserService() {
  print('\n🧪 Testing UserService...');
  
  final userService = UserService();
  
  // Test 1: Login Success
  final loginSuccess = userService.login('test@example.com', 'password123');
  TestFramework.expectTrue(loginSuccess, 'Login should succeed with correct credentials');
  
  // Test 2: Login Failure
  final loginFailure = userService.login('test@example.com', 'wrongpassword');
  TestFramework.expectFalse(loginFailure, 'Login should fail with wrong password');
  
  // Test 3: Register Success
  final registerSuccess = userService.register('newuser@example.com', 'newpassword');
  TestFramework.expectTrue(registerSuccess, 'Register should succeed with new email');
  
  // Test 4: Register Failure (duplicate)
  final registerFailure = userService.register('test@example.com', 'password123');
  TestFramework.expectFalse(registerFailure, 'Register should fail with existing email');
}

void testProductModel() {
  print('\n🧪 Testing Product Model...');
  
  // Test 1: Product Creation
  final product = Product(
    id: '1',
    name: 'wheat',
    category: 'Tahıl',
    description: 'Test product',
  );
  
  TestFramework.expect(product.id, '1', 'Product ID should be 1');
  TestFramework.expect(product.name, 'wheat', 'Product name should be wheat');
  
  // Test 2: Product toJson
  final json = product.toJson();
  TestFramework.expect(json['id'], '1', 'toJson should include correct ID');
  TestFramework.expect(json['name'], 'wheat', 'toJson should include correct name');
  
  // Test 3: Product fromJson
  final productFromJson = Product.fromJson(json);
  TestFramework.expect(productFromJson.id, '1', 'fromJson should create product with correct ID');
  TestFramework.expect(productFromJson.name, 'wheat', 'fromJson should create product with correct name');
}

void main() {
  print('🚀 TerraMind Frontend Tests Starting...');
  print('=' * 50);
  
  try {
    testProductService();
    testUserService();
    testProductModel();
    
    TestFramework.printResults();
    
    if (TestFramework._failedTests == 0) {
      print('\n🎉 All frontend tests passed!');
      print('✅ TerraMind frontend services are working correctly!');
      print('✅ Product service, User service, and Product model are functional!');
    }
    
  } catch (e) {
    print('❌ Test error: $e');
  }
}

File: frontend/test_simple_frontend.py
"""
En basit frontend testi - Python ile
Flutter/Dart yüklü olmadan da çalışır
"""

class TestFramework:
    """Basit test framework'ü"""
    
    def __init__(self):
        self.passed_tests = 0
        self.failed_tests = 0
    
    def expect(self, actual, expected, test_name):
        """Değer karşılaştırma testi"""
        if actual == expected:
            self.passed_tests += 1
            print(f'✅ {test_name}: PASSED')
        else:
            self.failed_tests += 1
            print(f'❌ {test_name}: FAILED')
            print(f'   Expected: {expected}')
            print(f'   Actual: {actual}')
    
    def expect_true(self, condition, test_name):
        """Boolean true testi"""
        if condition:
            self.passed_tests += 1
            print(f'✅ {test_name}: PASSED')
        else:
            self.failed_tests += 1
            print(f'❌ {test_name}: FAILED')
    
    def expect_false(self, condition, test_name):
        """Boolean false testi"""
        if not condition:
            self.passed_tests += 1
            print(f'✅ {test_name}: PASSED')
        else:
            self.failed_tests += 1
            print(f'❌ {test_name}: FAILED')
    
    def print_results(self):
        """Test sonuçlarını yazdır"""
        print('\n' + '=' * 50)
        print('📊 TEST RESULTS:')
        print(f'✅ Passed: {self.passed_tests}')
        print(f'❌ Failed: {self.failed_tests}')
        print(f'📈 Total: {self.passed_tests + self.failed_tests}')
        print('=' * 50)
        
        if self.failed_tests == 0:
            print('🎉 All tests passed!')
        else:
            print('⚠️ Some tests failed!')


class Product:
    """Product model sınıfı"""
    
    def __init__(self, id, name, category, description):
        self.id = id
        self.name = name
        self.category = category
        self.description = description
    
    def to_dict(self):
        """Dictionary'e çevir"""
        return {
            'id': self.id,
            'name': self.name,
            'category': self.category,
            'description': self.description,
        }
    
    @classmethod
    def from_dict(cls, data):
        """Dictionary'den oluştur"""
        return cls(
            id=data['id'],
            name=data['name'],
            category=data['category'],
            description=data['description']
        )


class ProductService:
    """Product service sınıfı"""
    
    def __init__(self):
        self._products = [
            Product('1', 'wheat', 'Tahıl', 'Temel besin maddesi olan tahıl ürünü'),
            Product('2', 'corn', 'Tahıl', 'Yüksek verimli tahıl ürünü'),
            Product('3', 'cotton', 'Endüstriyel', 'Tekstil endüstrisi için lif ürünü'),
        ]
    
    def get_all_products(self):
        """Tüm ürünleri getir"""
        return self._products.copy()
    
    def get_product_by_id(self, product_id):
        """ID'ye göre ürün getir"""
        for product in self._products:
            if product.id == product_id:
                return product
        return None
    
    def search_products(self, query):
        """Ürün ara"""
        query_lower = query.lower()
        return [
            product for product in self._products
            if query_lower in product.name.lower() or query_lower in product.category.lower()
        ]
    
    def get_products_by_category(self, category):
        """Kategoriye göre ürünleri getir"""
        return [product for product in self._products if product.category == category]


class UserService:
    """User service sınıfı"""
    
    def __init__(self):
        self._users = {
            'test@example.com': 'password123',
            'admin@example.com': 'admin123',
        }
    
    def login(self, email, password):
        """Kullanıcı girişi"""
        if email in self._users and self._users[email] == password:
            return True
        return False
    
    def register(self, email, password):
        """Kullanıcı kaydı"""
        if email in self._users:
            return False  # User already exists
        self._users[email] = password
        return True


def test_product_service():
    """ProductService testleri"""
    print('\n🧪 Testing ProductService...')
    
    test_framework = TestFramework()
    product_service = ProductService()
    
    # Test 1: GetAllProducts
    all_products = product_service.get_all_products()
    test_framework.expect(len(all_products), 3, 'GetAllProducts should return 3 products')
    
    # Test 2: GetProductById
    product = product_service.get_product_by_id('1')
    test_framework.expect_true(product is not None, 'GetProductById should return a product')
    if product:
        test_framework.expect(product.name, 'wheat', 'Product name should be wheat')
    
    # Test 3: SearchProducts
    search_results = product_service.search_products('wheat')
    test_framework.expect(len(search_results), 1, 'SearchProducts should return 1 result for wheat')
    
    # Test 4: GetProductsByCategory
    grain_products = product_service.get_products_by_category('Tahıl')
    test_framework.expect(len(grain_products), 2, 'GetProductsByCategory should return 2 grain products')
    
    return test_framework


def test_user_service():
    """UserService testleri"""
    print('\n🧪 Testing UserService...')
    
    test_framework = TestFramework()
    user_service = UserService()
    
    # Test 1: Login Success
    login_success = user_service.login('test@example.com', 'password123')
    test_framework.expect_true(login_success, 'Login should succeed with correct credentials')
    
    # Test 2: Login Failure
    login_failure = user_service.login('test@example.com', 'wrongpassword')
    test_framework.expect_false(login_failure, 'Login should fail with wrong password')
    
    # Test 3: Register Success
    register_success = user_service.register('newuser@example.com', 'newpassword')
    test_framework.expect_true(register_success, 'Register should succeed with new email')
    
    # Test 4: Register Failure (duplicate)
    register_failure = user_service.register('test@example.com', 'password123')
    test_framework.expect_false(register_failure, 'Register should fail with existing email')
    
    return test_framework


def test_product_model():
    """Product Model testleri"""
    print('\n🧪 Testing Product Model...')
    
    test_framework = TestFramework()
    
    # Test 1: Product Creation
    product = Product('1', 'wheat', 'Tahıl', 'Test product')
    test_framework.expect(product.id, '1', 'Product ID should be 1')
    test_framework.expect(product.name, 'wheat', 'Product name should be wheat')
    
    # Test 2: Product to_dict
    product_dict = product.to_dict()
    test_framework.expect(product_dict['id'], '1', 'to_dict should include correct ID')
    test_framework.expect(product_dict['name'], 'wheat', 'to_dict should include correct name')
    
    # Test 3: Product from_dict
    product_from_dict = Product.from_dict(product_dict)
    test_framework.expect(product_from_dict.id, '1', 'from_dict should create product with correct ID')
    test_framework.expect(product_from_dict.name, 'wheat', 'from_dict should create product with correct name')
    
    return test_framework


def main():
    """Ana test fonksiyonu"""
    print('🚀 TerraMind Frontend Tests Starting...')
    print('=' * 50)
    
    try:
        # Testleri çalıştır
        product_service_tests = test_product_service()
        user_service_tests = test_user_service()
        product_model_tests = test_product_model()
        
        # Toplam sonuçları hesapla
        total_passed = (product_service_tests.passed_tests + 
                       user_service_tests.passed_tests + 
                       product_model_tests.passed_tests)
        
        total_failed = (product_service_tests.failed_tests + 
                       user_service_tests.failed_tests + 
                       product_model_tests.failed_tests)
        
        # Sonuçları yazdır
        print('\n' + '=' * 50)
        print('📊 OVERALL TEST RESULTS:')
        print(f'✅ Passed: {total_passed}')
        print(f'❌ Failed: {total_failed}')
        print(f'📈 Total: {total_passed + total_failed}')
        print('=' * 50)
        
        if total_failed == 0:
            print('🎉 All frontend tests passed!')
            print('✅ TerraMind frontend services are working correctly!')
            print('✅ Product service, User service, and Product model are functional!')
            print('✅ Frontend architecture is ready for Flutter development!')
        else:
            print('⚠️ Some frontend tests failed!')
        
    except Exception as e:
        print(f'❌ Test error: {e}')


if __name__ == '__main__':
    main()
